### ISSUE TITLE
Auth0 Tenant data source doesn't return custom domains

### ISSUE THREAD

**Issue by anthonyangel** [auth0/terraform-provider-auth0#511]
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/terraform-provider-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/terraform-provider-auth0/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/auth0/terraform-provider-auth0/releases/latest) of this provider and the issue still persists.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

`data "auth0_tenant" "my_tenant" {}` returns an attribute `domain`, which is always the Auth0 domain name

### Expectation

If a custom domain name has been set then I would like the data source to return it

### Reproduction

```
resource "auth0_custom_domain" "my_custom_domain" {
  domain = "auth.example.com"
  type   = "auth0_managed_certs"
}

data "auth0_tenant" "my_tenant" {}

output "domain" {
  value = data.auth0_tenant.my_tenant.domain
}
```

### Auth0 Terraform Provider version

0.44.0

### Terraform version

1.3.9

**Comment by sergiught** [comment]
Hey @anthonyangel 👋🏻 

The `domain` property on the `auth0_tenant` data source will always be the Auth0 Tenant domain name as you can also check from our docs: https://registry.terraform.io/providers/auth0/auth0/latest/docs/data-sources/tenant#read-only. It was never intended to return the custom domain.

However we could consider adding an `auth0_custom_domain` data source to fit your needs. Something like:

```hcl
data "auth0_custom_domains" "my_domain" {}
```

That would fetch all the custom domains assigned to a tenant and have them available as a list within the terraform config. Usually a tenant will have only 1 custom domain set, however there's the possibility of having more than one (different subscription plans). 

The attributes of each item would be the same as the ones available for the custom domain resource: https://registry.terraform.io/providers/auth0/auth0/latest/docs/resources/custom_domain. 

Would this work for you?

**Comment by sergiught** [comment]
Hey @anthonyangel additional could you also give a bit more insights into the workflow that you're trying to accomplish by having access to the custom domains? Just want to fully understand your problem and see if there are perhaps even better ways to tackle it. Thanks!

**Comment by anthonyangel** [comment]
> Hey @anthonyangel additional could you also give a bit more insights into the workflow that you're trying to accomplish by having access to the custom domains? Just want to fully understand your problem and see if there are perhaps even better ways to tackle it. Thanks!

We have a Terraform workspace which sets up our core Auth0 config, which includes the custom domain (it is all the centralized platform config that is set once)

We use Auth0 organizations, mapping each organization to our tenants (it gets confusing when our definition of a tenant doesn't match the Auth0 definition). The config for this is set up via Terraform in a per-tenant workspace. We grab the custom domain and set it in an AWS SSM parameter for our app to use.

**Comment by sergiught** [comment]
Hey @anthonyangel   👋🏻 

Auth0 Custom Domain Data source is available in the latest release [v0.45.0](https://github.com/auth0/terraform-provider-auth0/releases/tag/v0.45.0).  Please give it a go and let us know if you have any issues with it. Thanks!

You can use it like so:

```
data auth0_custom_domain my_domain { }
```

Docs: https://registry.terraform.io/providers/auth0/auth0/0.45.0/docs
